### PR TITLE
Do not fail the deep test check on in progress runs

### PR DESCRIPTION
**Pull request by keyboardDrummer**
If there is a deep test check, and multiple runs qualify for that check, but the most recent run in still in progress, do not fail the check on that run, but instead look at the first not in progress run.

Also add a missing return statement so we don't get don't confusing error reporting like this:
image

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

**Comment by fabiomadge** [comment]
Doesn't this mean that the test always run as long as there is a deep test in progress? You should still make sure that the previous run passed.

**Comment by keyboardDrummer** [comment]
> Doesn't this mean that the tests always run as long as there is a deep test in progress? You should still make sure that the previous run passed.

You're right, thanks! Updated the code.

File: .github/workflows/check-for-workflow-run.js [hunk 1]
```diff
@@ -15,9 +15,10 @@ module.exports = async ({github, context, core, workflow_id, sha, ...config}) =>
   // run for this SHA we see.
   const runFilterDesc = sha ? `${workflow_id} on ${sha}` : workflow_id
   for (const run of result.data.workflow_runs) {
-    if (!sha || run.head_sha == sha) {
-      if (run.conclusion != "success") {
+    if ((!sha || run.head_sha === sha) && run.conclusion !== "in_progress") {
+      if (run.conclusion !== "success") {
         core.setFailed(`Last run of ${runFilterDesc} did not succeed: ${run.html_url}`)
+        return
```
Comment (fabiomadge) [inline .github/workflows/check-for-workflow-run.js:21]: Can you pull that out?

**Review by fabiomadge — APPROVED** [review]

File: .github/workflows/check-for-workflow-run.js (hunk 1 above)
Comment (keyboardDrummer) [reply .github/workflows/check-for-workflow-run.js:21]: I will, but I wonder whether it's worth the extra build trip ^,^

**Pull request merged**